src/entry.rs

```diff
@@ -631,9 +635,11 @@ impl<'a> EntryFields<'a> {
                 })?;
             }
         }
+        #[cfg(not(target_os = "wasi"))]
         if let Ok(mode) = self.header.mode() {
             set_perms(dst, Some(&mut f), mode, self.preserve_permissions)?;
         }
+        #[cfg(not(target_os = "wasi"))]
```

Comment:
Like above I don't think a `#[cfg]` here is necessary, I think it suffices if the default for` unpack_xattrs` is `false` for wasi.

Reply:
In the end this actually wasnt needed at all given the default implementation on WASM is just to return Ok(()). I removed this attribute